File: 29-06-2024_Identical_Linked_Lists.java
class Solution {
    // Function to check whether two linked lists are identical or not.
    public boolean areIdentical(Node head1, Node head2) {

        Node temp1 = head1, temp2 = head2;
        // traversing through both the linked lists.
        while (temp1 != null && temp2 != null) {

            // if data for current node in both lists aren't equal,
            // we return false.
            if (temp1.data != temp2.data) return false;

            temp1 = temp1.next;
            temp2 = temp2.next;
        }

        // if both lists reach end after same number of traversals
        // then true is returned else false.
        return (temp1 == null && temp2 == null);
    }
}

File: 29-06-2024_Identical_Linked_Lists.cpp
// Function to check whether two linked lists are identical or not
bool areIdentical(struct Node *head1, struct Node *head2) {
    // traversing through both the linked lists.
    while (head1 != NULL && head2 != NULL) {
        // if data for current node in both lists are not equal, we return false.
        if (head1->data != head2->data)
            return false;
        head1 = head1->next;
        head2 = head2->next;
    }
    // if both lists reach end after same number of traversals
    // then true is returned else false.
    return (head1 == NULL && head2 == NULL);
}
